keras/src/visualization/plot_image_gallery.py

```diff
@@ -0,0 +1,165 @@
+
+    return sample
+
+
+@keras_export("keras.visualization.plot_image_gallery")
+def plot_image_gallery(
+    images,
+    rows=None,
+    cols=None,
+    value_range=(0, 255),
+    scale=2,
+    path=None,
+    show=None,
+    transparent=True,
+    dpi=60,
+    legend_handles=None,
+    data_format=None,
+):
+    """Displays a gallery of images.
+
+    Args:
+        images: A 4D tensor or NumPy array of images. Shape should be
+           `(batch_size, height, width, channels)`.
+        value_range: A tuple specifying the value range of the images
+            (e.g., `(0, 255)` or `(0, 1)`). Defaults to `(0, 255)`.
+        rows: The number of rows in the gallery. If `None`, it's calculated
+            based on the number of images and `cols`. Defaults to `None`.
+        cols: The number of columns in the gallery. If `None`, it's calculated
+            based on the number of images and `rows`. Defaults to `None`.
+        scale: A float controlling the size of the displayed images. The images
+            are scaled by this factor. Defaults to `2`.
+        path: The path to save the generated gallery image. If `None`, the
+            image is displayed using `plt.show()`. Defaults to `None`.
+        show: Whether to display the image using `plt.show()`. If `True`, the
+            image is displayed. If `False`, the image is not displayed.
+            Ignored if `path` is not `None`. Defaults to `True` if `path`
+            is `None`, `False` otherwise.
+        transparent:  A boolean, whether to save the figure with a transparent
+            background. Defaults to `True`.
+        dpi: The DPI (dots per inch) for saving the figure. Defaults to 60.
+        legend_handles: A list of matplotlib `Patch` objects to use as legend
+            handles. Defaults to `None`.
+        data_format: The image data format `"channels_last"` or
+            `"channels_first"`. Defaults to the Keras backend data format.
+
+    Raises:
+        ValueError: If both `path` and `show` are set to non-`None` values or if
+            `images` is not a 4D tensor or array.
+        ImportError: if matplotlib is not installed.
+    """
+    if plt is None:
+        raise ImportError(
+            "The `plot_image_gallery` function requires the `matplotlib` "
+            "package. Please install it with `pip install matplotlib`."
+        )
+
+    if path is not None and show:
+        raise ValueError(
+            "plot_gallery() expects either `path` to be set, or `show` "
+            "to be true."
+        )
+
+    show = show if show is not None else (path is None)
+    data_format = data_format or backend.image_data_format()
+
+    batch_size = ops.shape(images)[0] if len(ops.shape(images)) == 4 else 1
+
+    rows = rows or int(math.ceil(math.sqrt(batch_size)))
+    cols = cols or int(math.ceil(batch_size // rows))
+    num_images = rows * cols
+
+    images = _extract_image_batch(images, num_images, batch_size)
```

Comment:
Hey @miticollo,

Thanks for pointing out this, I think we can remove this `raise ValueError` from `_extract_image_batch` and as per your suggestion we can expand dimension when image is just 3D tensor.

```python3
if len(ops.shape(images)) == 3: 
  images  = ops.expand_dims(images, 0)
batch_size = ops.shape(images)[0]
```

And the rest of the logic should hold good.